src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -430,8 +437,10 @@ public void submit(Runnable r)
     //logger.debug("{}.{}.{}", currentThread, r, eventThread);
     if (eventThread == currentThread && tasks.isEmpty()) {
       r.run();
-    }
-    else {
+    } else {
+      if (!isActive()) {
```

Comment:
Yes, I'd like to prevent adding tasks after the event loop has stopped but it is hard to distinguish the case when a task is added before the event loop is started from the case where it stopped. Note that in the second case the event loop may be restarted.